django/db/models/fields/__init__.py

```diff
@@ -304,6 +314,22 @@ def _check_choices(self):
             )
         ]
 
+    def _check_db_default(self):
+        if self.db_default is NOT_PROVIDED:
+            return []
+
+        if hasattr(self, 'model') and 'supports_functions_in_defaults' in self.model._meta.required_db_features:
+            # If the database does not support functions in defaults
+            # and the model is explicitly marked as requiring that feature,
+            # blocking management commands by returning an Error is unhelpful.
+            return []
+
+        try:
+            connection.ops.check_field_default_support(self.db_default)
+        except NotSupportedError as error:
+            return [checks.Error(error.args[0], obj=self, id='fields.E011')]
```

Comment:
I'm not sure whether `Error` or `Warning` is more appropriate here.

Reply:
`Error` makes more sense to me.